docs/vault/guide.md

```diff
@@ -113,11 +116,99 @@ To mint with your own (or another specific) Vault, you can use the Polkadot.js i
+
+Check the [Prometheus exporter](#querying-the-prometheus-exporter) section to find out how to query all the currently exposed metrics.
+
+
+Bridge-specific metrics (Collateralization, Locked Collateral, Required Collateral) get updated on each `FeedValues` oracle event, every ~25 mins. These are quite costly to update as they call parachain RPCs.
+
+### Running instructions
+
+Vault monitoring is enabled by default. The client provides the following configuration flags, identical to the ones used for monitoring Substrate nodes:
+```bash
+--no-prometheus
+   Do not expose a Prometheus metric endpoint
+--prometheus-external
+   Expose Prometheus exporter on all interfaces.
+   Default is local.
+--prometheus-port <PROMETHEUS_PORT>
+   Specify Prometheus exporter TCP Port
+   [default: 9615]
+```
+
+If monitoring is enabled, the client will log the metrics endpoint on startup. Below is an example:
+```
+./vault \
+--bitcoin-rpc-url http://localhost:18332 \
+--keyfile keyfile.json \
+--keyname "0x0e5aabe5ff862d66bcba0912bf1b3d4364df0eeec0a8137704e2c16259486a71" \
+--auto-register-with-faucet-url 'https://api-testnet.interlay.io/faucet' \
+--btc-parachain-url 'wss://api-testnet.interlay.io:443/parachain' \
+--prometheus-external
+...
+Mar 23 14:25:20.451  INFO vault: Starting Prometheus exporter at http://0.0.0.0:9615
+```
+
+#### Querying the Prometheus exporter
+To fetch the exported metrics, query the `/metrics` endpoint exposed by the Prometheus exporter with the command below. This will show all the available metrics.
+```bash
+curl http://HOST_IP:9615/metrics
+```
+
+#### Running Prometheus
+Configuration files for the Prometheus service are provided in the [interbtc-clients repo](https://github.com/interlay/interbtc-clients/tree/master/vault/src/monitoring).
+
+To customize the configuration of the Prometheus service, edit the `prometheus.yml` file using the command below.
+```bash
+git clone https://github.com/interlay/interbtc-clients
+cd interbtc-clients
+vim vault/src/monitoring/prometheus.yml
+```
+
+```bash
+git clone https://github.com/interlay/interbtc-clients
+cd interbtc-clients/vault/src/monitoring
+docker build -t prometheus .
+docker run -p 9090:9090 --network=host prometheus
+```
+
+#### Running Grafana
+If the default Prometheus port is used (`9615`), the default instructions from the Grafana docs will work by default: https://grafana.com/docs/grafana/latest/installation/debian
+
+Once Grafana is up and running, [import](https://grafana.com/docs/grafana/latest/dashboards/export-import/#import-dashboard) the [Vault client configuration](../_assets/config/grafana.json) file to see the metrics.
+
+### Example Visualisation
+![Vault Client Grafana Dashboard](../_assets/img/vault/granafa_monitoring.png)
+
+### Questions monitoring aims to answer
+
+#### How can I know if my client is offline?
```

Comment:
if the entire vault client is down, it would detect that. So we could leave this in as is for now and improve in a follow-up PR since we need to build more logic.